sql/slave.cc

```diff
@@ -618,7 +618,10 @@ int init_slave()
 
   /* If server id is not set, start_slave_thread() will say it */
 
-  if (active_mi->host[0] && !opt_skip_slave_start)
+  if (opt_skip_slave_start)
```

Comment:
There’re two usages of `opt_skip_slave_start`, one here, one in `Master_info_index::init_all_master_info(void)` @ `sql/rpl_mi.cc`.

Though the only call to `init_all_master_info()` is from this `init_slave(void)` itself at 3 dozens lines above here.
Looks like `init_slave` only initializes the `Master_info` for `""` and uses that to initialize named multi-source `Master_info`s as listed from `master.info.index` *(whatever that is)*.

This `if` group is the last part of this function, so logging here should be timely enough.